crates/validator-transaction-pool/src/lib.rs

```diff
@@ -27,104 +22,125 @@ impl TransactionFilter {
     }
 }
 
-pub type PullNotificationSender = aptos_channel::Sender<(), Arc<ValidatorTransaction>>;
-pub type PullNotificationReceiver = aptos_channel::Receiver<(), Arc<ValidatorTransaction>>;
+impl Default for TransactionFilter {
+    fn default() -> Self {
+        Self::PendingTxnHashSet(HashSet::new())
+    }
+}
 
-/// Create a validator txn pool for a given list of topics.
-/// For each topic, an optional notification sender can be specified,
-/// which is used to send a notification later when a pull on the topic happens.
-///
-/// Return a read client (typically used by consensus when proposing blocks).
-/// Return the write clients (typically used by validator transaction producers like DKG).
-pub fn new(
-    topic_tx_pairs: Vec<(Topic, Option<PullNotificationSender>)>,
-) -> (ReadClient, Vec<SingleTopicWriteClient>) {
-    let topics: Vec<Topic> = topic_tx_pairs.iter().map(|(topic, _)| *topic).collect();
-    let pool_state = Arc::new(Mutex::new(PoolState::new(topic_tx_pairs)));
-    let read_client = ReadClient {
-        pool: pool_state.clone(),
-    };
-    let write_clients = topics
-        .into_iter()
-        .map(|topic| SingleTopicWriteClient {
-            pool: pool_state.clone(),
-            topic,
-        })
-        .collect();
-    (read_client, write_clients)
+pub type VTxnPoolWrapper = Arc<Mutex<PoolState>>;
```

Comment:
idiomatic way is to define the outer as PoolState, inner as PoolStateInner with Arc plus Mutex

Reply:
fixed